### DISCUSSION TITLE
processus de deploiement

### DISCUSSION THREAD

**Discussion by mis7ern9o** [DstMlOpsCrypto/MainCrypto#26] [General]
Si nous partons sur la proposition de strructure de repo issue de la [discussion:Structure repo pour projet ML Ops #13](https://github.com/DstMlOpsCrypto/MainCrypto/discussions/13), nous avons :+1: 

├── maincrypto
│ ├── fastapi/
│ │-----|----test/
│ ├── mlflow/
│ │-----|----test/
│ ├── data-integration/
│ │-----|----test/
│ ├── db/
│ │-----|----test/

**Principe à discuter :** 

1. chaque dev est responsable de son repertoire applicatif (sela peut changer !)
2. le developer est responsable des lots de tests a executer
3. la librairie pytest est utilisée pour les tests
4. les tests sont executés en local avant de faire la PR
5. si les tests sont concluants l'application (folder) est deployer par github action (worker) sur les VM a partir du docker file présent dans chaque reperoire applicatif

**Points potentiellements problématiques**

- les EC2 change d'IP periodiquement (fermeture automatique). C'est gênant !
- acces au VM à partir de Github actions ?
- deploiement sur VM possible (droits compte aws limité) ?
- les secrets (VM) seront stocké ssur github (pas ds le repos !!) . Ceci est payant pour les repos privés !
- les tests réseaux ne pourront pas être réalisés (accés applicatifs...) car notre compte aws (a priori)  a des droits limités. Si c'était le cas nous utiliserions terraform !!

J'ai pu générer un github actions  en mode brutal avec tout ce dont j'explique. a paufiner bien sur.

Q3u'en pensez vous ?

**Comment by mis7ern9o** [comment]
il faut aussi gérer le "deploiement en local "docker" pour les tests

**Comment by mis7ern9o** [comment]
@DstMlOpsCrypto/members , j'ai vu un repo "master" !! il faudrait merger sur le main !!

**Reply by FredericLAGNIEZ** [reply]
On en a parlé lors de notre permière réunion, c'est un repo à supprimer, mais je ne peux pas le faire sur Github

**Reply by mis7ern9o** [reply]
repo master supprimé
